src/main/java/jnr/ffi/provider/jffi/DirectMemoryIO.java

```diff
@@ -186,8 +187,42 @@ public String getString(long offset) {
 
 
     public String getString(long offset, int maxLength, Charset cs) {
-        final byte[] bytes = IO.getZeroTerminatedByteArray(address() + offset, maxLength);
-        return cs.decode(ByteBuffer.wrap(bytes)).toString();
+    	long baseAddress = address() + offset;
+    	int nullTermSize = StringUtil.terminatorWidth(cs);
+        if(nullTermSize == 1) {
```

Comment:
Formatting throughout should match the rest of the code, which is intended to match typical Java coding conventions:

* four space indentation for blocks of code, 8-space indentation for line continuation
* space between keywords like `if` and `while` and their parenthesized conditionals
* spaces around operators like `+` and `%`

It's nitpicky but if we don't maintain consistent code formatting then we end up with future commits and PRs that have lots of unrelated changes.

Reply:
agree, will take care of that